src/core/xref.js

```diff
@@ -143,6 +128,43 @@ class XRef {
       }
       warn(`XRef.parse - Invalid "Root" reference: "${ex}".`);
     }
+
+    if (encrypt instanceof Dict) {
+      // Check if only the file attachments are encrypted.
+      if (
+        encrypt.get("CF")?.get("StdCF")?.get("AuthEvent")?.name === "EFOpen"
+      ) {
+        let hasEncryptedAttachments = false;
+        if (root instanceof Dict) {
+          const names = root.get("Names");
```

Comment:
I don't understand why you're doing that.
If there's no attachment then the user cannot click on them and the encryption stuff will never be used.
And if there are attachments then we've to only decrypt them when the user request to read them.
There is a bug (we talked about with @timvandermeij) about having to only load (and decrypt) the attachments when the user requests them, because right now we get them all the time.